tests/test_interpolation.py

```diff
@@ -419,3 +617,381 @@ def test_resolve_key_and_root(
+    ("dict_bool_key", "${test:{true: true, false: 'false'}}", GrammarParseError),
+    ("empty_dict_list", "${test:[],{}}", [[], {}]),
+    (
+        "structured_mixed",
+        "${test:10,str,3.14,true,false,inf,[1,2,3], 'quoted', \"quoted\", 'a,b,c'}",
+        [
+            10,
+            "str",
+            3.14,
+            True,
+            False,
+            math.inf,
+            [1, 2, 3],
+            "quoted",
+            "quoted",
+            "a,b,c",
+        ],
+    ),
+    (
+        "structured_deep",
+        "${test:{null0: [0, 3.14, false], true1: {a: [0, 1, 2], b: {}}}}",
+        {"null0": [0, 3.14, False], "true1": {"a": [0, 1, 2], "b": {}}},
+    ),
+    # Chained interpolations.
+    ("null_chain", "${null}", None),
+    ("true_chain", "${true}", True),
+    ("int_chain", "${int}", 123),
+    ("list_chain_bad_1", "${${prim_list}.0}", GrammarParseError),
+    ("dict_chain_bad_1", "${${prim_dict}.a}", GrammarParseError),
+    ("prim_list_copy", "${prim_list}", OmegaConf.create([-1, "a", 1.1])),
+    ("prim_dict_copy", "${prim_dict}", OmegaConf.create({"a": 0, "b": 1})),
+    ("list_chain", "${prim_list_copy.0}", -1),
+    ("dict_chain", "${prim_dict_copy.a}", 0),
+    # Nested interpolations.
+    ("ref_prim_str", "prim_str", ...),
+    ("nested_simple", "${${ref_prim_str}}", "hi"),
+    ("plans", {"plan A": "awesome plan", "plan B": "crappy plan"}, ...),
+    ("selected_plan", "plan A", ...),
+    (
+        "nested_dotted",
+        r"I choose: ${plans.${selected_plan}}",
+        "I choose: awesome plan",
+    ),
+    ("nested_deep", "${test:${${test:${ref_prim_str}}}}", "hi"),
+    ("nested_resolver", "${${test_str}:a, b, c}", ["a", "b", "c"]),
+    (
+        "nested_resolver_combined_illegal",
+        "${te${test_str_partial}:a, b, c}",
+        GrammarParseError,
+    ),
+    ("nested_args", "${test:${prim_str}, ${null}, ${int}}", ["hi", None, 123]),
+    # Relative interpolations.
+    (
+        "relative",
+        {"foo": 0, "one_dot": "${.foo}", "two_dots": "${..prim_dict.b}"},
+        OmegaConf.create({"foo": 0, "one_dot": 0, "two_dots": 1}),
+    ),
+    ("relative_nested", "${test:${.relative.foo}}", 0),
+    # Unmatched braces.
+    ("missing_brace_1", "${test:${prim_str}", GrammarParseError),
+    ("missing_brace_2", "${${test:prim_str}", GrammarParseError),
+    ("extra_brace", "${test:${prim_str}}}", "hi}"),
+]
+
+
+@pytest.mark.parametrize(  # type: ignore
+    "key,expected",
+    [
+        pytest.param(
+            key, _maybe_create(definition) if expected is ... else expected, id=key
+        )
+        for key, definition, expected in TEST_CONFIG_DATA
+    ],
+)
+def test_all_interpolations(restore_resolvers: Any, key: str, expected: Any) -> None:
+    dbg_test_access_only = False  # debug flag to not test against expected value
+    os.environ["OMEGACONF_TEST_ENV_INT"] = "123"
+    os.environ.pop("OMEGACONF_TEST_MISSING", None)
```

Comment:
Use monkeypatch.env (iirc) to set environment in tests.

Reply:
Done in ad00d79